### PR TITLE
Added test for RedirectMigrate

### PR DESCRIPTION
**Pull request by dantleech**
Functional tests for https://github.com/symfony-cmf/RoutingAuto/pull/54

This also fixes some tests which were false positives.

### CODE REVIEW

File: Tests/Functional/EventListener/AutoRouteListenerTest.php [hunk 1]
```diff
@@ -376,14 +376,17 @@ public function testLeaveRedirect($data, $updatedData, $expectedRedirectRoutePat
...
+            $this->assertNotNull($redirectRoute, 'Redirect exists for: ' . $originalPath);
             $this->assertEquals(AutoRouteInterface::TYPE_REDIRECT, $redirectRoute->getDefault('type'));
         }
 
         foreach ($expectedAutoRoutePaths as $newPath) {
             $autoRoute = $this->getDm()->find(null, $newPath);
-            $this->assertNotNull($redirectRoute, 'Autoroute exists for: ' . $originalPath);
-            $this->assertEquals(AutoRouteInterface::TYPE_REDIRECT, $redirectRoute->getDefault('type'));
+            if (null === $autoRoute) {
+                var_dump($newPath);die();;
```
Comment (dantleech) [inline Tests/Functional/EventListener/AutoRouteListenerTest.php]: Remove

File: Tests/Functional/EventListener/AutoRouteListenerTest.php [hunk 2]
```diff
@@ -376,14 +376,17 @@ public function testLeaveRedirect($data, $updatedData, $expectedRedirectRoutePat
...
+            }
+            $this->assertNotNull($autoRoute, 'Autoroute exists for: ' . $newPath);
+            $this->assertEquals(AutoRouteInterface::TYPE_PRIMARY, $autoRoute->getDefault('type'));
```
Comment (dantleech) [inline Tests/Functional/EventListener/AutoRouteListenerTest.php]: Before we were not even checking that the new route had been created.

**Comment by dantleech** [comment]
This works, but there are 2 problems:

- We are using `@String` in the fixtures -- so the PHP7 tests fail. I guess we need to update the ODM requirement to 1.3.
- The tests fail with the minimum requirements for the Routing component. I guess this should also be upgraded?

/cc @dbu @WouterJ

**Comment by dantleech** [comment]
Created an issue for fix the tests. Since the tests are already failing and this PR is working on all builds except PHP7 and minimum-requirements feel free to merge it. //cc @WouterJ 

https://github.com/symfony-cmf/RoutingAutoBundle/issues/161

**Comment by WouterJ** [comment]
Tests on master pass now, I'm fine with merging this PR.

**Comment by dantleech** [comment]
Rebased.

**Pull request merged**